src/classes/AFFL_MultiRecordType_TDTM.cls

```diff
@@ -152,8 +135,7 @@ public class AFFL_MultiRecordType_TDTM extends TDTM_Runnable {
             if (oldlist != null && oldlist.size() > 0) {
                 for (SObject so : oldlist) {
                     Affiliation__c afflOld = (Affiliation__c)so;
-                    String lookupFieldLabel = afflMapper.accRecTypeToContactLabel.get(afflOld.Affiliation_Type__c);
-                    String lookupFieldAPIName = afflMapper.contactLabelNames.get(lookupFieldLabel);
+                    String lookupFieldAPIName = afflMapper.getPrimaryAffiliationLookupAPIFromAffiliationType(afflOld.Affiliation_Type__c);
```

Comment:
slight logic change to fix the bug in scenario 3 (testing notes and quip doc).

Reply:
This is what it should have been like all along.